src/java.base/share/classes/jdk/internal/constant/PrimitiveClassDescImpl.java

```diff
@@ -40,6 +40,33 @@
 public final class PrimitiveClassDescImpl
         extends DynamicConstantDesc<Class<?>> implements ClassDesc {
 
+    /** {@link ClassDesc} representing the primitive type {@code int} */
+    public static final PrimitiveClassDescImpl CD_int = new PrimitiveClassDescImpl("I");
+
+    /** {@link ClassDesc} representing the primitive type {@code long} */
+    public static final PrimitiveClassDescImpl CD_long = new PrimitiveClassDescImpl("J");
+
+    /** {@link ClassDesc} representing the primitive type {@code float} */
+    public static final PrimitiveClassDescImpl CD_float = new PrimitiveClassDescImpl("F");
+
+    /** {@link ClassDesc} representing the primitive type {@code double} */
+    public static final PrimitiveClassDescImpl CD_double = new PrimitiveClassDescImpl("D");
+
+    /** {@link ClassDesc} representing the primitive type {@code short} */
+    public static final PrimitiveClassDescImpl CD_short = new PrimitiveClassDescImpl("S");
+
+    /** {@link ClassDesc} representing the primitive type {@code byte} */
+    public static final PrimitiveClassDescImpl CD_byte = new PrimitiveClassDescImpl("B");
+
+    /** {@link ClassDesc} representing the primitive type {@code char} */
+    public static final PrimitiveClassDescImpl CD_char = new PrimitiveClassDescImpl("C");
+
+    /** {@link ClassDesc} representing the primitive type {@code boolean} */
+    public static final PrimitiveClassDescImpl CD_boolean = new PrimitiveClassDescImpl("Z");
+
+    /** {@link ClassDesc} representing the primitive type {@code void} */
+    public static final PrimitiveClassDescImpl CD_void = new PrimitiveClassDescImpl("V");
```

Comment:
Since all the `PrimitiveClassDescImpl` instances are now created in this class, the constructor can probably be made `private`.